owncloudApp/src/main/java/com/owncloud/android/presentation/ui/sharing/fragments/ShareFileComposeFragment.kt

```diff
@@ -0,0 +1,626 @@
+                        chosenNumber = current + 1
+                        break
+                    }
+                }
+                if (chosenNumber < 0) {
+                    chosenNumber = usedNumbers[usedNumbers.size - 1] + 1
+                }
+            }
+
+            return defaultName + String.format(
+                Locale.getDefault(),
+                DEFAULT_NAME_SUFFIX, chosenNumber
+            )
+        }
+
+    private val ocCapabilityViewModel: OCCapabilityViewModel by viewModel {
+        parametersOf(
+            account?.name
+        )
+    }
+
+    private val ocShareViewModel: OCShareViewModel by viewModel {
+        parametersOf(
+            file?.remotePath,
+            account?.name
+        )
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        arguments?.let {
+            file = it.getParcelable(ARG_FILE)
+            account = it.getParcelable(ARG_ACCOUNT)
+        }
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        val shareWithUsersAllowed = resources.getBoolean(R.bool.share_with_users_feature)
+        val shareViaLinkAllowed = resources.getBoolean(R.bool.share_via_link_feature)
+        val shareWarningAllowed = resources.getBoolean(R.bool.warning_sharing_public_link)
+        return ComposeView(requireContext()).apply {
+            filterTouchesWhenObscured = PreferenceUtils.shouldDisallowTouchesWithOtherVisibleWindows(requireContext())
+            setContent {
+                setViewCompositionStrategy(ViewCompositionStrategy.DisposeOnViewTreeLifecycleDestroyed)
+
+                // Get capabilities to update some UI elements depending on them
+                val capabilitiesState = ocCapabilityViewModel.capabilities.observeAsState()
+                val capabilitiesUiResult = capabilitiesState.value?.peekContent()
+                val capabilities = capabilitiesUiResult?.getStoredData()
+                val isShareApiEnabled = capabilities?.filesSharingApiEnabled == CapabilityBooleanType.TRUE
+                val isPublicShareEnabled = capabilities?.filesSharingPublicEnabled == CapabilityBooleanType.TRUE
+                val isMultiplePublicSharingEnabled = capabilities?.filesSharingPublicMultiple?.isTrue ?: false
+
+                // Get shares to update some UI elements depending on them
+                val sharesState = ocShareViewModel.shares.observeAsState()
+                val sharesUiResult = sharesState.value?.peekContent()
+                val shares = sharesUiResult?.getStoredData()
+                var privateShares = shares?.filter { share ->
+                    share.shareType == ShareType.USER ||
+                            share.shareType == ShareType.GROUP ||
+                            share.shareType == ShareType.FEDERATED
+                }
```

Comment:
We should always try to write immutability variables when possible. That way, you won't need those ugly `!!` later 🙀 , and you will find lesser unexpected states.

Also, when I need to show the same result for a null or empty list, I normally try to remove nullability and fall back to an empty list and check just `isEmpty` instead of `isNullOrEmpty` and avoid some `?.`

```suggestion
                val privateShares = shares?.filter { share ->
                    share.shareType == ShareType.USER ||
                            share.shareType == ShareType.GROUP ||
                            share.shareType == ShareType.FEDERATED
                }?.sortedBy { it.sharedWithDisplayName } ?: emptyList()
```

Let me know what u think!

Reply:
Much better, I wasn't sure if that ``sortedBy`` would cause effect once the state is updated, but it seems to have sense so yep, I will definitely change it 👍